### ISSUE TITLE
Using with SOS2 causes pawns invisible when walking in vacuum

### ISSUE THREAD

**Issue by hayashi-tsukigin** [vegapnk/RJW-Genes#120]
RJW Genes are thought to be the most likely cause of the bug by Visual Exceptions

```
Exception in Verse.TickList.Tick: System. NullReferenceException: Object reference not set to an instance of an object
[Ref C2820596]
at RJW_Genes.GeneUtility. OffsetLifeForce (RimWorld. IGeneResourceDrain drain, System. Single offset) [0x00000] in <15637142d2484a599d9836c99c68f62f>:0
at RJW_Genes.Gene_LifeForce_Empath. FarmLifeForce (Verse. Pawn farmedPawn) [0x0004f] in <15637142d2484a599d9836c99c68f62f>:0
at RJW_Genes.Gene_LifeForce_Empath. Tick () [0x00051] in <15637142d2484a599d9836c99c68f62f>:0
at RimWorld.Pawn_GeneTracker. GeneTrackerTick () [0x0003f] in <f0ac5eb9b52e4cc396c70fc9a4ee15e5>:0
at Verse.Pawn.Tick () [0x00678] in <f0ac5eb9b52e4cc396c70fc9a4ee15e5>:0
- TRANSPILER Krkr.RocketMan.Soyuz: IEnumerable`1 Soyuz.Patches. Pawn_Tick_Patch:Transpiler(IEnumerable`1 instructions, ILGenerator generator)
- PREFIX RIMMSqol: Boolean RIMMSqol.ThinkResultContextRecover:Prefix()
- POSTFIX alt4s. alliesarehelpful: Void PawnTendAndRescuePatch:Postfix(Pawn __instance)
- POSTFIX XeoNovaDan. VisiblePants: Void VisiblePants. Patch_Pawn+Patch_Tick:Postfix(Pawn __instance)
- POSTFIX Roolo. DualWield: Void DualWield.HarmonyInstance. Pawn_Tick:Postfix(Pawn __instance)
- FINALIZER Krkr.RocketMan.Soyuz: Void Soyuz.Patches. Pawn_Tick_Patch:Finalizer(Pawn __instance, Exception __exception)
at Verse.TickList.Tick () [0x0017d] in <f0ac5eb9b52e4cc396c70fc9a4ee15e5>:0
- TRANSPILER net.pardeike.rimworld.lib.harmony: IEnumerable`1 VisualExceptions. ExceptionsAndActivatorHandler:Transpiler(IEnumerable`1 instructions, MethodBase original)
```

**Comment by vegapnk** [comment]
@hayashi-tsukigin Thanks for your report! 
I also want to do an SOS2 run soon, so this is very helpful. 

Does this happen once, or does this throw a lot of errors at once? 

Is the LifeForceEmpath getting invisible or the other pawn?

**Comment by hayashi-tsukigin** [comment]
> 感谢您的报告！我也想尽快运行 SOS2，所以这很有帮助。
> 
> 这是发生一次，还是一次抛出很多错误？
> 
> 是 LifeForceEmpath 变得隐形还是另一个棋子？

When the archive was first loaded, there were no errors, and pawns could walk normally in space without being invisible.

After a few minutes, the error was thrown once, after which pawns became invisible in the vacuum.

Since most of my pawns are all of one genotype (this is due to the xenogenetic seeds of Warhammer 40K), they all have LifeForceEmpath, so they will all be invisible in the vacuum.

The only pawn I don't have this gene is SOS2's core hologram, which I should test to see if she's invisible when I get back.

**Comment by hayashi-tsukigin** [comment]
SOS2's core hologram does not contain this bug, indicating that this bug is caused by LifeForceEmpath

**Comment by vegapnk** [comment]
[Relevant File](https://github.com/vegapnk/RJW-Genes/blob/main/Source/Genes/Life_Force/Genes/Gene_LifeForce_Empath.cs) 

[This Could be a missing null-check](https://github.com/vegapnk/RJW-Genes/blob/a685209dc18e4223bf719623e1d45fa1a06b4fa7/Source/Genes/Life_Force/Genes/Gene_LifeForce_Empath.cs#L73) for pawns that have no agetracker. 

[This](https://github.com/vegapnk/RJW-Genes/blob/a685209dc18e4223bf719623e1d45fa1a06b4fa7/Source/Genes/Life_Force/Genes/Gene_LifeForce_Empath.cs#L97) could be another missing null-check if the empath and his lifeforce is null for whatever reason. 

[The GeneUtility](https://github.com/vegapnk/RJW-Genes/blob/a685209dc18e4223bf719623e1d45fa1a06b4fa7/Source/Genes/GeneUtility.cs#L18) also has no null-checks, maybe the input float is null for whatever reason.

**Comment by vegapnk** [comment]
Should be fixed with v2.2.1
